internals/cli/cmd_check.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2025 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package cli
+
+import (
+	"fmt"
+
+	"github.com/canonical/go-flags"
+	"gopkg.in/yaml.v3"
+
+	"github.com/canonical/pebble/client"
+)
+
+const cmdCheckSummary = "Query the details of a configured health check"
+const cmdCheckDescription = `
+The check command shows details for a single check in YAML format.
+`
+
+type cmdCheck struct {
+	client *client.Client
+
+	Positional struct {
+		Check string `positional-arg-name:"<check>" required:"1"`
+	} `positional-args:"yes"`
+}
+
+func init() {
+	AddCommand(&CmdInfo{
+		Name:        "check",
+		Summary:     cmdCheckSummary,
+		Description: cmdCheckDescription,
+		New: func(opts *CmdOptions) flags.Commander {
+			return &cmdCheck{client: opts.Client}
+		},
+	})
+}
+
+func (cmd *cmdCheck) Execute(args []string) error {
+	if len(args) > 0 {
+		return ErrExtraArgs
+	}
+
+	opts := client.ChecksOptions{
+		Names: []string{cmd.Positional.Check},
+	}
+	checks, err := cmd.client.Checks(&opts)
+	if err != nil {
+		return err
+	}
+	if len(checks) == 0 {
+		fmt.Fprintln(Stderr, "No matching health checks.")
```

Comment:
Because they're asking for a single check by name, I think this should be an error (which will return nonzero). Similar to `cmdIdentity`:

```go
		return fmt.Errorf("cannot find check %q", cmd.Positional.Check)
```